### ISSUE TITLE
CMake Error: No source or binary directory provided

### ISSUE THREAD

**Issue by ulyssesrr** [RadeonOpenCompute/ROCT-Thunk-Interface#28]
```shell
$ make package-dev
VERSION BUILD DEFINED 2.0.0-0-g3ba20ce
-- Configuring done
-- Generating done
-- Build files have been written to: /home/ulysses/git/ROCT-Thunk-Interface/build
CMake Error: No source or binary directory provided
make[3]: *** [CMakeFiles/build-dev.dir/build.make:57: CMakeFiles/build-dev] Error 1
make[2]: *** [CMakeFiles/Makefile2:142: CMakeFiles/build-dev.dir/all] Error 2
make[1]: *** [CMakeFiles/Makefile2:181: CMakeFiles/package-dev.dir/rule] Error 2
make: *** [Makefile:225: package-dev] Error 2
```

I have tracked the issue to the build-dev custom target command:

```cmake
add_custom_target ( build-dev
    COMMAND ${CMAKE_COMMAND}
        -DSOURCE_DIR="${SOURCE_DIR}"
        -DCMAKE_INSTALL_PREFIX="${CMAKE_INSTALL_PREFIX}"
        -DCPACK_RPM_PACKAGE_REQUIRES="hsakmt-roct"
        -DCPACK_PACKAGING_INSTALL_PREFIX="${CPACK_PACKAGING_INSTALL_PREFIX}"
    WORKING_DIRECTORY ${DEV_BUILD_DIR} )
```
It invokes cmake with some vars arguments `-Dvar=value` but without a directory argument.

It seems this was an undocummeted behaviour of cmake that worked by accident and became broken on cmake-3.13.3 : https://gitlab.kitware.com/cmake/cmake/issues/18817
Acording to the link above they will revert to a warning on 3.13.4 but its expected to become an error again in cmake 3.14.

The fix is to add a directory argument, just adding the current directory worked for me, since the command specifies a WORKING_DIRECTORY:
https://gist.github.com/ulyssesrr/d7c93e121bc9c5c7a4e0564be5e4ac21

**Comment by rigred** [comment]
Thanks for posting this - also coincidentally helped me fix an issue on yet another project aside from just roct.

**Comment by kentrussell** [comment]
Fixed in 3.1 (well, fixed before then, but not an issue now)
